### ISSUE TITLE
Private key / JSON logs into wrong address. Private key public address mismatch

### ISSUE THREAD

**Issue by bluecollarcrypto** [kvhnuke/etherwallet#1121]
Hello,

When creating the wallet, I copied and pasted public address & private key into .txt file and saved on my pc. 
I saved JSON file on my pc. 
I type password into txt. file which gets saved to my pc, from which I copy and paste password into browser field when creating new wallet. 

Private key opens incorrect address.
JSON file opens incorrect address.

View only via public address opens & loads correct wallet address and displays my REAL tokens.

I would like to transfer my REAL tokens and can not right now

**Comment by kvhnuke** [comment]
It seems like you saved the wrong private key or a private key for a different wallet. Does the private key and JSON file derive the same address?

**Comment by bluecollarcrypto** [comment]
I'm not sure that is the case because I save at moment of creation prior to execute any transfers etc. 
The private key and JSON should derive from same address as it was created and saved all upon creating the new wallet. 

Im hoping I do not have issue as posted here
https://www.reddit.com/r/ethereum/comments/48rt6n/using_myetherwalletcom_just_burned_me_for/

Additionally I am wondering if token issuer may be able to recall tokens and transfer them to other address I use as it was an ICO event for these tokens I am only able to view at this time

**Comment by kvhnuke** [comment]
Does your JSON file and private key derive the same address?

**Comment by bluecollarcrypto** [comment]
How do I check that ?

**Comment by kvhnuke** [comment]
Open the wallet using your private key, then remember the address you see. Then unlock the wallet using the JSON file and check whether that address matches the previous one. Also let me know what token you are trying to transfer?

**Comment by bluecollarcrypto** [comment]
It opens incorrect wallet with JSON or private key. Only able to view with public address. I saved JSON and private at time of wallet creation

**Comment by kvhnuke** [comment]
Are the public address shown by the private key and JSON the same?

**Comment by bluecollarcrypto** [comment]
Let me log in with each and see

**Comment by bluecollarcrypto** [comment]
JSON and Private log ins do bring up same public address

**Comment by kvhnuke** [comment]
Ok, what kind of token are you trying to transfer? And how did you receive these tokens? From an exchange? ICO?

**Comment by bluecollarcrypto** [comment]
ICO. It was for REAL. Here is my public address. I created this wallet for this specific ICO as I do for other ICOs (to not show all other balances I hold)

0x07E9Cd8DbC4186014DBcaa1520eB648F98F87919

**Comment by kvhnuke** [comment]
upon checking your address it seems like you definitely had access to the correct privatekey. you were able to send out multiple transactions approx 41 days ago. I would go through all other private keys you have and see whether any of those derives the correct address.

**Comment by bluecollarcrypto** [comment]
Let me see what I come up with

**Comment by kvhnuke** [comment]
Check under MonethaTokenCrowdsale it seems like you used that address for that initially

**Comment by bluecollarcrypto** [comment]
Yes I did. I was unsuccessful, it sold in minutes. I noted that as well in my ICO wallet and participation notes. Let me look to see if i can find more in monetha notes

**Comment by tayvano** [comment]
Did you ever unlock your wallet by hand typing the key?

Please be sure that you are carefully and accurately typed it. A single wrong character will open a different wallet than "yours".

If you go to https://www.myetherwallet.com/helpers.html, scroll allllll the way to the bottom, and enter your private key and address and see if that resolves. If that doesn't work, try hand typing your private key faster or slower or with your eyes closed. 

Essentially we are trying to recreate the way you accessed the wallet the first time to try to regain access.

**Comment by bluecollarcrypto** [comment]
Hi Taylor,

Thank you for continued follow up. 
I only access my MEW(s) via copy and paste (which is saved info upon creating and into text file in local machine) 

Also I've only used my one computer and one internet connection method for all my crypto activities. Cookies ? or maybe some other type of analysis could prove this out. 

The wallet I can not access was only created for a specific ICO. Only holds small eth and those tokens from ICO, just want to send those tokens to my primary MEW coupled with Trezor I use to hold all my other alt coins. 

As we speak that ICO and the CEO of that company,  who I have been corresponding with, is offering 50% of my 6388 tokens (in wallet I can't access) to me as a gift of good faith and generosity out of their reserve, into wallet I can access. It is very very much appreciated. 

It may not be possible, I'll ask anyway - if anyway on your end to send my 6388 tokens back to the company that distributed them... that way I could possibly have 100% my tokens and the company's reserve would not be affected and used for better cause. 

Thank you for your time and consideration 
Brandon

**Comment by iquidus** [comment]
It's not possible for the MEW team to move coins from any address, the server/nodes never see any private keys or wallet files, MEW does not hold any coins. Everything takes place clientside in your browser.

**Comment by Uelese** [comment]
I've had the same problem. I generated a new MEW for the purpose of connecting into EtherDelta. MEW generates a private key which is automatically saved to my computer using a surface 3. Once i unlocked my new wallet, I scrolled down and copied the address from that wallet and sent coins from my MEW which is connected to my trezor. I checked the transaction and it had been successful, so I went back to my private key which was saved in my comp with the name UTC-2018 etc and none of the 3 keys in that file match the original address which was generated. My mistake is that I didn't close and double check the keys matched but I'm in the same boat as many others. I'm hoping for a solution which i understand is slim. Any help appreciated.

Thanks

**Comment by wparad** [comment]
@Uelese, could you expand with more details, i.e. the address you copied, tx adress, and the actions you performed others than those listed.  It also sounds like you are saying that the file you opened show two different addresses on each subsequent opening, how is that possible?

**Comment by Uelese** [comment]
@wparad this is the address that the wallet generated "0x04eb66d8cbdc5f822aba31c6bd4790519c098228" TX is "0xf28c171b9abb54a77e4c6de5b55bc4ccaa10f252e402c5fa44b268c4bc8170b5" I opened my wallet using the private key option. I then opened the file which saves automatically when creating the wallet. In order to read the file i converted it into notepad so i could get the numbers, and cut out the numbers which were bracketed right after the "cipher text", and it didn't match the address i had sent it too. I then scrolled across the same file and saw some numbers after "salt": - So i cut those out and entered them into the private key option on MEW and still not the same. I scrolled across the same file and some numbers right after the "mac": and cut them out and still not the same. That's how I got 3 different addresses from that file. I then opened my wallet using the "Keystore/ JSON file and it brings up the one address which is not the same one i copied to send my tokens too.

**Comment by wparad** [comment]
@Uelese, I'm confused, two things of note:
* there are many more than 3 values in the JSON keystore.  As a matter of fact the public address is none of three you listed but actually the ones saved in the `address` field.
* looking at address [0x04eb66d8cbdc5f822aba31c6bd4790519c098228](https://etherscan.io/address/0x04eb66d8cbdc5f822aba31c6bd4790519c098228#tokentxns), seems you managed to successful sign two transactions after the transfer from your trezor from this address to this [address](https://etherscan.io/address/0xeb36367cd402c3756c6cbe3ba0bb1cde4312008e#tokentxns). 

This happened after your last post, does that mean you figured this out?

**Comment by Uelese** [comment]
HI Warren,

Yes, I had a friend come over last night and work on it with me.
Thank you for looking into this, I have no idea what had happened but I'm
just glad I got my coins out.

I also asked my mate what the process was but he didn't know what he did to
get me to that address as we had been playing with MEW for hours to get in.
Thank you again, Warren.

Cheers
Uelese

**Comment by numero41** [comment]
@Uelese 

Please can you ask your friend to give us an update about what he did, because there are other people that try to figure out the same issues.
Thank you!

@bluecollarcrypto

**Comment by numero41** [comment]
@bluecollarcrypto 

Is your issue similar to this one?
https://github.com/kvhnuke/etherwallet/issues/1528

**Comment by Massat168** [comment]
I have same problem.

When I created MEW account, I copied, screen shotted the private key, and downloaded the keystore file, but both private key and keystore file gives me different account.

I have already put my coins in my account, so I cant take my coin out....

Does anyone know how to fix it?
